### PR TITLE
Scroll inertia cancel for macOS

### PR DESCRIPTION
**Pull request by moffatman**
Generate PointerScrollInertiaCancel event on macOS when the trackpad receives a touch during momentum scrolling. 

Part of https://github.com/flutter/flutter/issues/106979

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by chinmaygarde** [comment]
cc @gspencergoog

File: shell/platform/darwin/macos/framework/Source/FlutterViewController.mm [hunk 1]
```diff
@@ -825,4 +837,31 @@ - (void)swipeWithEvent:(NSEvent*)event {
   // Not needed, it's handled by scrollWheel.
 }
 
+- (void)touchesBeganWithEvent:(NSEvent*)event {
+  if (@available(macOS 10.12, *)) {
+    NSTouch* touch = event.allTouches.anyObject;
```
Comment (dkwingsmt) [inline shell/platform/darwin/macos/framework/Source/FlutterViewController.mm:842]: Can we make sure this contains either 0 or 1 touch? Maybe assert it?

**Review by dkwingsmt — APPROVED** [review]
LGTM with comments

File: shell/platform/darwin/macos/framework/Source/FlutterViewController.mm (hunk 1 above)
Comment (moffatman) [reply shell/platform/darwin/macos/framework/Source/FlutterViewController.mm:842]: The number of touches shouldn't matter, though. If the user places two fingers simultaneously on the trackpad, we should still cancel inertia.

**Comment by chinmaygarde** [comment]
@dkwingsmt This looks good to go. Can we land this please?

**Pull request merged**